# HG changeset patch
# User Betterbird <betterbird@betterbird.eu>
# Date 1744234420 -7200
# Parent  eaa8a8d64a7004186be81c18d39ba910b2c870f8
Feature: Implement pref mail.tabs.firstTab.startupURI.

diff --git a/mail/app/profile/all-thunderbird.js b/mail/app/profile/all-thunderbird.js
--- a/mail/app/profile/all-thunderbird.js
+++ b/mail/app/profile/all-thunderbird.js
@@ -581,8 +581,9 @@ pref("mail.tabs.tabMinWidth", 100);
 pref("mail.tabs.tabMaxWidth", 250);
 pref("mail.tabs.tabClipWidth", 140);
 pref("mail.tabs.autoHide", true);
 pref("mail.tabs.closeWindowWithLastTab", true);
+pref("mail.tabs.firstTab.startupURI", "");
 
 // Allow the tabs to be in the titlebar on supported systems
 pref("mail.tabs.drawInTitlebar", true);
 
diff --git a/mail/base/content/about3Pane.js b/mail/base/content/about3Pane.js
--- a/mail/base/content/about3Pane.js
+++ b/mail/base/content/about3Pane.js
@@ -630,8 +630,20 @@ var folderPaneContextMenu = {
     this._checkMenuItem(
       "folderPaneContext-favoriteFolder",
       flags & Ci.nsMsgFolderFlags.Favorite
     );
+    // --- Set up startup folder menu item.
+    if (!isServer) {
+      const startup = Services.prefs.getStringPref(
+        "mail.tabs.firstTab.startupURI",
+        ""
+      );
+      this._checkMenuItem(
+        "folderPaneContext-startupFolder",
+        folder.URI == startup
+      );
+    }
+    this._showMenuItem("folderPaneContext-startupFolder", !isServer);
     this._showMenuItem("folderPaneContext-markAllFoldersRead", isServer);
 
     this._showMenuItem("folderPaneContext-settings", isServer);
     this._showMenuItem("folderPaneContext-filters", isServer);
@@ -889,8 +901,23 @@ var folderPaneContextMenu = {
         break;
       case "folderPaneContext-sendUnsentMessages":
         goDoCommand("cmd_sendUnsentMsgs");
         break;
+      case "folderPaneContext-startupFolder": {
+        const startup = Services.prefs.getStringPref(
+          "mail.tabs.firstTab.startupURI",
+          ""
+        );
+        let newval;
+        if (folder.URI == startup) {
+          // Is the folder is already the startup folder, we toggle.
+          newval = "";
+        } else {
+          newval = folder.URI;
+        }
+        Services.prefs.setStringPref("mail.tabs.firstTab.startupURI", newval);
+        break;
+      }
       case "folderPaneContext-markAllFoldersRead":
         topChromeWindow.MsgMarkAllFoldersRead([folder]);
         break;
       case "folderPaneContext-settings":
diff --git a/mail/base/content/about3Pane.xhtml b/mail/base/content/about3Pane.xhtml
--- a/mail/base/content/about3Pane.xhtml
+++ b/mail/base/content/about3Pane.xhtml
@@ -479,8 +479,11 @@
     <menuitem id="folderPaneContext-favoriteFolder"
               type="checkbox"
               label="&folderContextFavoriteFolder.label;"
               accesskey="&folderContextFavoriteFolder.accesskey;"/>
+    <menuitem id="folderPaneContext-startupFolder"
+              type="checkbox"
+              label="&folderContextStartupFolder.label;"/>
     <menuitem id="folderPaneContext-properties"
               class="menuitem-iconic"
               label="&folderContextProperties2.label;"
               accesskey="&folderContextProperties2.accesskey;"/>
diff --git a/mail/base/content/mailTabs.js b/mail/base/content/mailTabs.js
--- a/mail/base/content/mailTabs.js
+++ b/mail/base/content/mailTabs.js
@@ -179,8 +179,22 @@ var mailTabType = {
             tab.chromeBrowser.contentWindow.paneLayout.messagePaneVisible,
         };
       },
       restoreTab(tabmail, persistedState) {
+        // Note that tabs are persisted in session.json.
+        // For the first tab we just ignore the persisted value.
+        if (persistedState.firstTab) {
+          const startup = Services.prefs.getStringPref(
+            "mail.tabs.firstTab.startupURI",
+            ""
+          );
+          if (startup) {
+            const f = MailUtils.getExistingFolder(startup);
+            if (f) {
+              persistedState.folderURI = f.URI;
+            }
+          }
+        }
         if (!persistedState.firstTab) {
           tabmail.openTab("mail3PaneTab", persistedState);
           return;
         }
diff --git a/mail/locales/en-US/chrome/messenger/messenger.dtd b/mail/locales/en-US/chrome/messenger/messenger.dtd
--- a/mail/locales/en-US/chrome/messenger/messenger.dtd
+++ b/mail/locales/en-US/chrome/messenger/messenger.dtd
@@ -624,8 +624,9 @@
 <!ENTITY folderContextProperties2.label "Properties">
 <!ENTITY folderContextProperties2.accesskey "P">
 <!ENTITY folderContextFavoriteFolder.label "Favorite Folder">
 <!ENTITY folderContextFavoriteFolder.accesskey "a">
+<!ENTITY folderContextStartupFolder.label "Startup Folder">
 <!ENTITY folderContextSettings2.label "Settings">
 <!ENTITY folderContextSettings2.accesskey "e">
 
 <!-- Search Bar -->
